### PR TITLE
[14.0][ADD] l10n_th_company_novat

### PR DESCRIPTION
**Pull request by kittiu**
Allow configure a company (usually governmental) to be a Not VAT Registered organization. And have following impacts,

* Prevent VAT in any document.
* But WHT amount can still be caucluated based on amount untaxed (for VAT registered vendor), and amount total (for No-VAT registered vendor)

### CODE REVIEW

**Comment by kittiu** [comment]
image

image

**Comment by kittiu** [comment]
This is sample case when Vendor is No-VAT registerd. Please take note.

image

**Comment by kittiu** [comment]
@newtratip @flook-chaiwat Please study and review this module.

**Comment by kittiu** [comment]
Ready for review.

File: l10n_th_company_novat/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Copyright 2020 Ecosoft Co., Ltd (https://ecosoft.co.th/)
```
Comment (newtratip) [inline l10n_th_company_novat/__manifest__.py]:
```suggestion
# Copyright 2021 Ecosoft Co., Ltd (https://ecosoft.co.th/)
```

File: l10n_th_company_novat/readme/CONFIGURATION.rst [hunk 2]
```diff
@@ -0,0 +1,14 @@
+Setup Company,
+
+* Go to menu, Settings > User & Companies > Companies
```
Comment (newtratip) [inline l10n_th_company_novat/readme/CONFIGURATION.rst]:
```suggestion
* Go to menu Settings > Users & Companies > Companies
```

File: l10n_th_company_novat/readme/CONFIGURATION.rst [hunk 3]
```diff
@@ -0,0 +1,14 @@
+Setup Company,
```
Comment (newtratip) [inline l10n_th_company_novat/readme/CONFIGURATION.rst]:
```suggestion
Setup Company
```

File: l10n_th_company_novat/readme/CONFIGURATION.rst [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+* Go to menu, Settings > User & Companies > Companies
+* Set up Not VAT registered = True
+
+Setup Partner,
```
Comment (newtratip) [inline l10n_th_company_novat/readme/CONFIGURATION.rst]:
```suggestion
Setup Partner
```

File: l10n_th_company_novat/readme/CONFIGURATION.rst [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+* Go to menu Partner
```
Comment (newtratip) [inline l10n_th_company_novat/readme/CONFIGURATION.rst]:
```suggestion
* Go to menu Contacts
```

File: l10n_th_company_novat/static/description/index.html [hunk 6]
```diff
@@ -0,0 +1,438 @@
+<?xml version="1.0" encoding="utf-8" ?>
```
Comment (newtratip) [inline l10n_th_company_novat/static/description/index.html]: Remove this file.

File: l10n_th_company_novat/models/partner.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+        return super().map_tax(taxes, product=product, partner=partner)
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    novat = fields.Boolean(
+        string="Not VAT Registered",
+        help="This partner is not a vat registered vendor/customer. "
+        "Odoo will try to remove taxes from document ",
```
Comment (newtratip) [inline l10n_th_company_novat/models/partner.py]:
```suggestion
        "Odoo will try to remove taxes from document",
```

File: l10n_th_company_novat/models/res_company.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    novat = fields.Boolean(
+        string="Not VAT registered",
```
Comment (newtratip) [inline l10n_th_company_novat/models/res_company.py]:
```suggestion
        string="Not VAT Registered",
```

File: l10n_th_company_novat/models/sale.py [hunk 9]
```diff
@@ -0,0 +1,9 @@
+# Copyright 2020 Ecosoft Co., Ltd (https://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+from odoo import models
+
+
+class SaleOrderLine(models.Model):
+    _name = "sale.order.line"
+    _inherit = ["sale.order.line", "base.company.novat"]
+    _tax_field_name = "taxes_id"
```
Comment (newtratip) [inline l10n_th_company_novat/models/sale.py]:
```suggestion
    _tax_field_name = "tax_id"
```

File: l10n_th_company_novat/models/account_move.py [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    _tax_field_name = "tax_ids"
+
+    wtvat = fields.Float(
+        string="Vat%",
+        compute="_compute_wtvat",
+        store=True,
+        readonly=False,
+        copy=True,
+        help="Only with No-VAT registered company, set default tax  "
+        "to calculate base amount used for withholding amount ",
```
Comment (newtratip) [inline l10n_th_company_novat/models/account_move.py]:
```suggestion
        "to calculate base amount used for withholding amount",
```

File: l10n_th_company_novat/models/account_move.py [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    @api.depends("move_id.partner_id")
+    def _compute_wtvat(self):
+        if not self.env.company.novat:
+            self.update({"wtvat": False})
+            return
+        for rec in self:
+            partner = rec.move_id.partner_id
+            percent = False
+            if partner and not partner.novat:  # VAT partner
+                move_type = self.move_id.move_type
```
Comment (newtratip) [inline l10n_th_company_novat/models/account_move.py]:
```suggestion
                move_type = rec.move_id.move_type
```

**Comment by kittiu** [comment]
All fixed. Thanks so much for your review.

**Comment by newtratip** [comment]
My a little comment
- Copyright should change from 2020 to 2021 of all *.py files
- index.html need to this PR. If not please remove it. https://github.com/OCA/l10n-thailand/blob/ac2c09cee6e434e239a05c4dc331b5320f24e160/l10n_th_company_novat/static/description/index.html

**Review by newtratip — APPROVED** [review]
Functional Review LGTM :+1:

**Comment by newtratip** [comment]
@kittiu, please attend my comment https://github.com/OCA/l10n-thailand/pull/231#issuecomment-830807529 too.

Thank you.

**Comment by kittiu** [comment]
One more review please.
@Saran440 @ps-tubtim @flook-chaiwat

**Review by Saran440 — APPROVED** [review]

**Pull request merged**